packages/dds/tree/src/test/memory/tree.spec.ts

```diff
@@ -95,63 +154,151 @@ describe("SharedTree memory usage", () => {
 					}
 				}
 
 				public beforeIteration(): void {
-					this.sharedTree = createLocalSharedTree("testSharedTree");
+					this.sharedTree = createLocalSharedTree(
+						"testSharedTree",
+						RootNodeSchema,
+						new RootNodeSchema(initialState),
+					);
 				}
 			})(),
 		).timeout(40000); // Set relatively higher threshold as 100_000 iterations can take a while.
 
 		benchmarkMemory(
 			new (class implements IMemoryTestObject {
 				public readonly title =
-					`Set an optional integer property ${x} times in a local SharedTree, then clear it`;
+					`Set an optional integer property ${numberOfEntries} times in a local SharedTree, then clear it`;
 				private sharedTree?: TreeView<typeof RootNodeSchema>;
 
 				public async run(): Promise<void> {
 					assert(this.sharedTree?.root.child !== undefined);
 
-					for (let i = 0; i < x; i++) {
-						this.sharedTree.root.child.propertyOne = x;
+					for (let i = 0; i < numberOfEntries; i++) {
+						this.sharedTree.root.child.propertyOne = numberOfEntries;
 					}
 					this.sharedTree.root.child.propertyOne = undefined; // This is possible since the property is optional.
 				}
 
 				public beforeIteration(): void {
-					this.sharedTree = createLocalSharedTree("testSharedTree");
+					this.sharedTree = createLocalSharedTree(
+						"testSharedTree",
+						RootNodeSchema,
+						new RootNodeSchema(initialState),
+					);
 				}
 			})(),
 		).timeout(40000); // Set relatively higher threshold as 100_000 iterations can take a while.
 	}
+
+	function runBenchmarkMemoryForSubTree<TSchema extends ImplicitFieldSchema>(
+		title: string,
+		schema: TSchema,
+		generateContent: (numberOfNodes: number) => InsertableTreeFieldFromImplicitField<TSchema>,
+		testNodeCounts: number[],
+	) {
+		describe(title, () => {
+			for (const numberOfNodes of testNodeCounts) {
+				for (const forestType of [ForestType.Reference, ForestType.Optimized]) {
+					benchmarkMemory(
+						new (class implements IMemoryTestObject {
+							public readonly title =
+								`initialize ${numberOfNodes} nodes into tree using ${forestType === 0 ? "ObjectForest" : "ChunkedForest"}`;
+
+							private sharedTree: TreeView<typeof schema> | undefined;
+
+							public async run(): Promise<void> {
+								this.sharedTree = createLocalSharedTree(
+									"testSharedTree",
+									schema,
+									generateContent(numberOfNodes),
+									{
+										forest: forestType,
+										treeEncodeType: TreeCompressionStrategy.Compressed,
+									},
+								);
+							}
+
+							public beforeIteration(): void {
+								this.sharedTree = undefined;
+							}
+						})(),
+					).timeout(400000);
+				}
+			}
+		});
+	}
+
+	const numberOfNodesForTests = isInPerformanceTestingMode ? [1, 10, 100, 1000] : [10];
+	// TODO: needs .only to describe block when running in performance mode if you need to see the results. Check to see why it does not run without .only.
```

Comment:
I think it would be good to file a bug to track this and reference it here

Reply:
created a task and added task number to TODO comment